zap/src/main/java/org/zaproxy/zap/extension/alert/AlertViewPanel.java

```diff
@@ -621,6 +633,19 @@ private String getSourceData(Alert alert) {
         return strBuilder.toString();
     }
 
+    /**
+     * @param alert
+     * @return a pretty string representing injection location or empty string if alert is injection
+     *     location is null/empty
+     */
+    private String getInjectionLocationText(Alert alert) {
+        if (alert.getInjectionLocation() == null || alert.getInjectionLocation().length() == 0) {
+            return "";
+        }
+        return Constant.messages.getString(
+                "variant.alert.location." + alert.getInjectionLocation());
```

Comment:
This will lead to errors with custom variants, should fallback to some name if the key is not present.